tmt/base.py

```diff
@@ -2130,6 +2130,22 @@ def lint_unknown_keys(self) -> LinterReturn:
 
         yield LinterOutcome.PASS, 'correct keys are used'
 
+    def lint_empty_env_files(self) -> LinterReturn:
+        """ P001: env files are not empty """
+
+        env_files = self.node.get("environment-file") or []
+        if len(env_files) > 0:
+            for env_file in env_files:
+                path = os.path.normpath(env_file)
+                size = os.path.getsize(path)
```

Comment:
Use `pathlib` instead of `os.path`, please. It will eventually make the condition simpler as well:

```python
for _env_file in env_files:
  env_file = Path(env_file).resolve()

  if not env_file.exists() or not env_file.stat().st_size:
    ..
```

Catches both cases when the file does not exists - deserves its own message - and the file being empty.

Reply:
Updated, thanks !